### PR TITLE
Remove dependency on `hashbrown`

### PR DESCRIPTION
**Pull request by tiziano88**
This makes the build non-deterministic:
https://github.com/project-oak/oak/issues/862#issuecomment-616519455

It was introduced in an attempt to make `oak_abi` `no_std` compatible,
but that's not necessary any more.

Ref #862

# Checklist

- [ ] Pull request affects core Oak functionality (e.g. runtime, SDK, ABI)
  - [ ] I have written tests that cover the code changes.
  - [ ] I have checked that these tests are run by [Cloudbuild](cloudbuild.yaml)
  - [ ] I have updated [documentation](docs/) accordingly.
  - [ ] I have raised an [issue](https://github.com/project-oak/oak/issues) to
        cover any TODOs and/or unfinished work.
- [ ] Pull request includes prototype/experimental work that is under
      construction.

### CODE REVIEW

**Comment by tiziano88** [comment]
@daviddrysdale I did `cd cargo && cargo raze`, but I don't understand what's going on, why did those deps change? And why do I not see any changes related to the dep I actually removed?

**Comment by tiziano88** [comment]
Ah never mind, I re-tried after updating `cargo/Cargo.toml` and now at least the negative diffs match what I would expect. Still not sure about all the others though. I note that I don't see any `cargo/Cargo.lock` file checked in, my guess is that `cargo-raze` is just pulling whatever latest versions are available right now?

**Comment by daviddrysdale** [comment]
Yeah, there should probably be a checked-in `cargo/Cargo.lock`.  Also, it's worth removing now-unused entries from `cargo/remote/*` &ndash; clearing the whole directory before running `cargo-raze` should do it.  #871 in progress

**Comment by tiziano88** [comment]
@googlebot ping

File: cargo/crates.bzl [hunk 1]
```diff
@@ -16,15 +16,6 @@ def _new_git_repository(name, **kwargs):
 
 def raze_fetch_remote_crates():
 
-    _new_http_archive(
-        name = "raze__ahash__0_3_2",
-        url = "https://crates-io.s3-us-west-1.amazonaws.com/crates/ahash/ahash-0.3.2.crate",
-        type = "tar.gz",
-        strip_prefix = "ahash-0.3.2",
-
-        build_file = Label("//cargo/remote:ahash-0.3.2.BUILD"),
```
Comment (daviddrysdale) [inline cargo/crates.bzl:25]: Just to clarify: I think these `cargo/remote/<removed-dep>.BUILD` files should be removed in this PR too.

Comment (tiziano88) [reply cargo/crates.bzl:25]: Done.

**Pull request merged**